### DISCUSSION TITLE
announce: mkdexe2

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#2235] [General]
https://github.com/dosemu2/mkdexe2
Does anyone want to try that out? :)

Ideally we also need the flatpak builds
on github, so that if dosemu2 is not
installed on host, the script could get
it in flatpak.

**Comment by stsp** [comment]
@probonopd have you considered
the fuse-overlayfs for writable overlays?
DOS progs are not well enough educated
to write to tmpdirs and so on... :(

**Reply by probonopd** [reply]
Yes: https://github.com/AppImage/AppImageKit/wiki/Bundling-Windows-applications#3-make-it-possible-to-use-a-read-only-wineprefix-1

**Reply by stsp** [reply]
Yes, works for me, thanks:
https://github.com/dosemu2/mkdexe2/commit/9faab706efd2dac9e7f618bd66e31276d6025b04

We probably still can't distribute the
AppImaged games for licensing reasons,
but for example @emendelson can now
package his WP with ready-to-use dosemu2
config, and completely remove all the
lengthy docs.
@nelgin can do the same with BBS games,
and this can also improve bug-reporting,
as now you can just provide an AppImage
to the report.

All lengthy docs should go away, things
should just work! And we are now much
closer to that state.

**Reply by nelgin** [reply]
Eh? it looks like I missed something?

**Comment by nelgin** [comment]
Looks interesting but I'm not sure I'd want to make use of it. First of all, a lot of DOS door games have a nightly maintenance task that is run. I just have a .bat script to do through and do all the scripts and be done with it. With this dexe thing it looks like I'd have to invoke the image and then figure how to run the maintenance. How would multiuser games be handled that require things like file locking. Most door games require some sort of configuration by the sysop to include their name, bbs, maybe a registration code. 

I can see how this would work well for a single user who wants to play DOS games on their system, but not in an active multiuser door type environment. I'm willing to be proved wrong, but my set up just works (barring the odd code change that breaks things ;) ), so I'm reluctant to change.

**Reply by jharrison03** [reply]
> Looks interesting but I'm not sure I'd want to make use of it. First of all, a lot of DOS door games have a nightly maintenance task that is run. I just have a .bat script to do through and do all the scripts and be done with it. With this dexe thing it looks like I'd have to invoke the image and then figure how to run the maintenance. How would multiuser games be handled that require things like file locking. Most door games require some sort of configuration by the sysop to include their name, bbs, maybe a registration code.
> 
> I can see how this would work well for a single user who wants to play DOS games on their system, but not in an active multiuser door type environment. I'm willing to be proved wrong, but my set up just works (barring the odd code change that breaks things ;) ), so I'm reluctant to change.

I think people capable of setting up a multi-user DOS BBS with doors is capable of building the project and customizing the setup appropriately.  This looks like it's just providing an additional way to install the project and isn't taking anything away so I don't see the harm in giving options.  In the case of a native Linux BBS launching DOS doors, they can launch the doors for the maintenance routine in the same way it calls out the doors for a user by passing the necessary parameters to DOSemu on the command line.  JMHO, of course.

**Reply by stsp** [reply]
> This looks like it's just providing an additional way to install the project

This actually provides the way to
run the project w/o ever installing.
And I think this is what he complains
about (not entirely sure thuogh).
I.e. if the maintenance work is done
with some host-base scripts, then indeed
the non-installing solution looks fishy.
But as he mentions .bat files, not the
host-based scripts, I don't think I actually
understand the problem.

**Comment by nelgin** [comment]
The problem is that my BBS, which runs on Linux, called maint.bat every night at midnight. This launches dosemu which then goes through and does the maintenance on each door.

```
e:
cd \tw2002
extern MULTITASK=NO
cd \shuffle
shufword shuffle.cfg /E
cd \godcrime
maint
cd \arcesc
maint
```

For example.

Sure, I could change this to invoke the dexe but it all seems like an unnecessary overhead for what I need. 

A couple of people have put a lot of time into creating a dosemu2 config that works with the BBS so stuff just works.  I have no problem installing the doors, I have a special dosemu that I can run to do that since I have to make config changes anyway.

Again, for a BBS, I don't think this would work well, not for me anyway, but if I just wanted to play some offline games like Doom or Leisure Suit Larry, then this would be great :)

**Reply by stsp** [reply]
But the last time I tried, there were
a lot of things to install under dosemu
itself. This was probably a "local mode"
though, correct?
So do you mean, in "remote mode" all
the setup is done on host, and dosemu
only opens a virtual comport, so there is
nothing to distribute?

**Comment by nelgin** [comment]
I created a dosemu2.tar.gz for Synchronet bbs sysops which they just drop on their system and it has everything they need. The proper start up files, drivers, dosemu.conf file, .dosemu directory and the like so it's very easy to get going. It's all contained within the /sbbs (or whatever root they have for their bbs) so it doesn't interfere with any other dosemu they may have on their system.

**Comment by stsp** [comment]
I am actually thinking if this mkdexe2
stuff deserves the rpm/deb packaging
or not. It seems too small for a package,
plus it downloads appimagetool from
github, as its not packaged either.

I'd be much more comfortable with
something pip-alike, so whoever wants
this, can have the per-user install.
But pip is probably only for python,
not for shell scripts.
Does anyone know some pip-alike
solution for shell scripts?
Or should that just be rpmed/debed?
